runtime/bcverify/vrfyhelp.c

```diff
@@ -931,8 +931,16 @@ isFieldAccessCompatible(J9BytecodeVerificationData *verifyData, J9ROMFieldRef *f
 				 * the field is accessible only when it exists in the subclass or the superclass's
 				 * initialization is completed in which case uninitializedThis is set to FALSE.
 				 */
-				J9BranchTargetStack *liveStack = (J9BranchTargetStack *)verifyData->liveStack;
-				return findFieldFromCurrentRomClass(romClass, fieldRef) || !liveStack->uninitializedThis;
+				J9ROMFieldShape *field = findFieldFromCurrentRomClass(romClass, fieldRef);
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+				if (J9_ARE_NO_BITS_SET(field->modifiers, J9AccStrict)) {
```

Comment:
There should be a NULL check on `field`. 
This should be changed to return `FALSE` if `J9AccStrict` is set. If strict flag is not set, the code should still go throw `isClassCompatibleByName()`.